tests/unit/Infrastructure/DashboardSamlBundle/Security/Authentication/Provider/SamlProviderTest.php

```diff
@@ -85,14 +89,85 @@ public function test_administrator_teams_validation_rejects_invalid_teams()
+        $this->attributeDictionary->shouldReceive('translate')->andReturn($adapter);
+        $adapter->shouldReceive('getNameID')->andReturn('does-not-exist-yet');
+        $adapter->shouldReceive('getAttributeValue')->with('mail')->andReturn(['john@example.com']);
+        $adapter->shouldReceive('getAttributeValue')->with('commonName')->andReturn(['John Doe']);
+        $adapter->shouldReceive('getAttributeValue')->with('isMemberOf')->andReturn([]);
+        $adapter->shouldReceive('getAttributeValue')->with('urn:mace:surfnet.nl:surfnet.nl:sab:role:SURFconext-verantwoordelijke', [])->andReturn(
+            [
+                'urn:mace:surfnet.nl:surfnet.nl:sab:organizationCode:ibuildings',
+                'urn:mace:surfnet.nl:surfnet.nl:sab:role:SURFconext-verantwoordelijk',
+            ]
+        );
+        $this->contactRepo->shouldReceive('findByNameId')->with('does-not-exist-yet')->andReturnNull();
+        $this->serviceRepo->shouldReceive('findByTeamNames')->andReturnNull();
+        $this->contactRepo->shouldReceive('save');
+        $user = $provider->getUser($assertion);
+        self::assertEquals($user->getRoles(), ['ROLE_SURFCONEXT_REPRESENTATIVE']);
+    }
+
+    public function test_authorization_surfconext_representative_and_user()
+    {
+        $provider = $this->buildProvider("'urn:collab:foo:team.foobar.com'", 'urn:mace:surfnet.nl:surfnet.nl:sab:role:SURFconext-verantwoordelijk');
+
+        $assertion = m::mock(Assertion::class);
+        $adapter = m::mock(AssertionAdapter::class);
+        $this->attributeDictionary->shouldReceive('translate')->andReturn($adapter);
+        $adapter->shouldReceive('getNameID')->andReturn('does-not-exist-yet');
+        $adapter->shouldReceive('getAttributeValue')->with('mail')->andReturn(['john@example.com']);
+        $adapter->shouldReceive('getAttributeValue')->with('commonName')->andReturn(['John Doe']);
+        $adapter->shouldReceive('getAttributeValue')->with('isMemberOf')->andReturn(['team-1', 'team-2']);
+        $adapter->shouldReceive('getAttributeValue')->with('urn:mace:surfnet.nl:surfnet.nl:sab:role:SURFconext-verantwoordelijke', [])->andReturn(
+            [
+                'urn:mace:surfnet.nl:surfnet.nl:sab:organizationCode:ibuildings',
+                'urn:mace:surfnet.nl:surfnet.nl:sab:role:SURFconext-verantwoordelijk',
+            ]
+        );
+        $service1 = m::mock(Service::class);
+        $service2 = m::mock(Service::class);
+        $this->serviceRepo->shouldReceive('findByTeamNames')->with(['team-1', 'team-2'])->andReturn([$service1, $service2]);
+        $this->contactRepo->shouldReceive('findByNameId')->with('does-not-exist-yet')->andReturnNull();
+        $this->serviceRepo->shouldReceive('findByTeamNames')->andReturnNull();
+        $this->contactRepo->shouldReceive('save');
+        $user = $provider->getUser($assertion);
+        self::assertEquals($user->getRoles(), ['ROLE_SURFCONEXT_REPRESENTATIVE', 'ROLE_USER']);
```

Comment:
What happens if the user has an extra rikenin the future? Then this test would fail I guess.

Isn't it better to check if the roles intersect?

Reply:
Im totally fine with having to update (and specify) this assertion if anything in authz changes in the future. That only shows that the behavior changes. And allows us to verify if the existing workings are not broken by the new change.